src/cluster_legacy.c

```diff
@@ -7049,7 +7049,36 @@ int clusterCommandSpecial(client *c) {
         clusterDelNode(n);
         clusterDoBeforeSleep(CLUSTER_TODO_UPDATE_STATE | CLUSTER_TODO_SAVE_CONFIG);
         addReply(c, shared.ok);
-    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && c->argc == 3) {
+    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && (c->argc == 3 || c->argc == 4)) {
+        /* CLUSTER REPLICATE (<NODE ID> | NO ONE)*/
+        if (c->argc == 4) {
+            /* CLUSTER REPLICATE NO ONE */
+            if (strcasecmp(c->argv[2]->ptr, "NO") != 0 || strcasecmp(c->argv[3]->ptr, "ONE") != 0) {
+                addReplySubcommandSyntaxError(c);
+                return 1;
+            }
+            if (nodeIsPrimary(myself)) {
+                addReply(c, shared.ok);
+                return 1;
+            }
+            sds client = catClientInfoShortString(sdsempty(), c, server.hide_user_data_from_log);
+            serverLog(LL_NOTICE, "Stop replication and turning myself into empty primary (request from '%s').", client);
+            sdsfree(client);
+            clusterSetNodeAsPrimary(myself);
+            clusterPromoteSelfToPrimary();
+            emptyData(-1, server.repl_replica_lazy_flush ? EMPTYDB_ASYNC : EMPTYDB_NO_FLAGS, NULL);
```

Comment:
I think we should also kill the child process.

```suggestion
            flushAllDataAndResetRDB(server.repl_replica_lazy_flush ? EMPTYDB_ASYNC : EMPTYDB_NO_FLAGS);
```

Reply:
I applied this change but it looks to me like unnecessary. ````"CLUSTER REPLICATE NO ONE"```` can be only called on replica nodes. Thus, there can't be any RDB in process.

BTW, why don't we do the same on ````"CLUSTER RESET"```` command? In opposite to ````"CLUSTER REPLICATE NO ONE"````, ````"CLUSTER RESET"```` can be called on primary.

Reply:
@PingXie 
I applied this change but it looks to me like unnecessary. ````"CLUSTER REPLICATE NO ONE"```` can be only called on replica nodes. Thus, there can't be any RDB in process.

BTW, why don't we do the same on ````"CLUSTER RESET"```` command? In opposite to ````"CLUSTER REPLICATE NO ONE"````, ````"CLUSTER RESET"```` can be called on primary.

Reply:
I think it is a coincidence that we don't use chained replication in the cluster mode but that is an orthogonal decision so it is still preferable imo to not take that dependency.

Yeah I think cluster reset should kill the rdb child process too.